Meter.c

```diff
@@ -286,14 +292,38 @@ static const char* const GraphMeterMode_dotsAscii[] = {
    /*20*/":", /*21*/":", /*22*/":"
 };
 
+static void GraphMeterMode_expandGraphData(GraphData* data, unsigned int numRecords) {
+   if (data->numRecords >= numRecords) {
+      assert(numRecords == 0 || data->values != NULL);
+      return;
+   }
+
+   double* buf = malloc(numRecords * sizeof(*data->values));
+   if (!buf)
+      return;
+
+   memset(buf, 0, (numRecords - data->numRecords) * sizeof(*data->values));
+   if (data->numRecords > 0) {
+      assert(data->values != NULL);
+      memcpy(buf + (numRecords - data->numRecords), data->values, data->numRecords * sizeof(*data->values));
+   }
+   free(data->values);
```

Comment:
This approach (manual malloc/copy/free instead of realloc) obviously works, but honestly it strikes me as a somewhat excessive (micro)optimization effort for a rarely-taken path -- AFAIK this is only going to be exercised while someone is actively resizing their terminal, which is a vanishingly tiny fraction of htop's runtime in any remotely normal scenario.

If we want to optimize something in the graph code, I think it would make a lot more sense to convert the values array into a circular buffer so we're not shuffling its entire contents over by one slot on every update -- while I suspect it'd still be pretty small difference in the scheme of things, that would at least be an O(n) to O(1) improvement that would actually be on a common-case code path.  (But hopefully as a separate, later change independent of this PR.)